drafts/own-graphql-spring-boot-starter.md

```diff
@@ -0,0 +1,271 @@
+
+```gradle
+implementation 'org.example:graphql-info-spring-boot-starter:1.0-SNAPSHOT'
+```
+
+With this setup, when the consuming application starts, it will have access to the `/graphql` endpoint with the `info` query automatically available.
+
+## 9. Query the GraphQL endpoint
+
+Start the application, open `http://localhost:8080/graphiql?path=/graphql` and check that the following query returns the hard-coded value `version=0.0.1`.
+
+```graphql
+query info {
+  info
+}
+```
+
+## 11. Configuratively set the version
+
+You can allow users of your starter to customize the endpoint behavior via `application.properties`. For example, you can inject version number from the configuration like this:
+
+`GraphQLStarterAutoConfiguration.java`:
+```java
+package org.example;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties(GraphQLStarterProperties.class)
+public class GraphQLStarterAutoConfiguration {
+
+    @Autowired
+    private GraphQLStarterProperties properties;
+
+    @Bean
+    @ConditionalOnMissingBean
+    public GraphQLController graphQLController() {
+        return new GraphQLController(properties);
+    }
+}
+```
+
+`GraphQLController.java`:
+```java
+package org.example;
+
+import org.springframework.graphql.data.method.annotation.QueryMapping;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class GraphQLController {
+
+    private final GraphQLStarterProperties properties;
+
+    public GraphQLController(GraphQLStarterProperties properties) {
+        this.properties = properties;
+    }
+
+    @QueryMapping
+    public String info() {
+        return "version=%s".formatted(this.properties.getVersion());
+    }
+}
+```
+
+`application.properties`:
+```properties
+info.app.version=1.0.0
+```
+
+The graphql query `info` no returns the configured value `1.0.0`.
+
+## 12. Add parameters
+
+Whe you add additional query or mutation methods to the starters `schema.graqphql` and `GraphQLController`,
+you might run into parameter-related errors. In that case, adding the `-parameters` compiler argument helps, 
+as described in the [Spring 6.x release notes]{https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x#parameter-name-retention}.
+
+``` kotlin
+tasks.withType<JavaCompile>() {
+    options.compilerArgs.add("-parameters")
+}
+```
+
+## 13. Troubleshooting
+
+To get more insight into the GraphQL schema loading and query mapping process, you can enable debug-level logging in the consuming applications `application.properties`:
+
+```properties
+logging.level.org.springframework.graphql=DEBUG
+```
+
```

Comment:
**suggestion:** I like when articles end with a short summary in 1-2 sentences and give some suggestions in which cases this could be used and what the next steps could be.

Reply:
added a short summary and a sentence about possible next steps.